crates/proof-of-sql-sdk-local/src/commitment_scheme.rs

```
@@ -44,11 +47,36 @@ pub trait CommitmentEvaluationProofId:
 {
     /// The [`CommitmentScheme`] associated with this commitment type.
     const COMMITMENT_SCHEME: CommitmentScheme;
+
+    /// Owned version of the verifier public setup.
+    type OwnedVerifierPublicSetup;
+
+    /// Deserialize the verifier public setup from bytes.
+    fn deserialize_verifier_setup(bytes: &[u8]) -> Result<Self::OwnedVerifierPublicSetup, String>;
 }
 
 impl CommitmentEvaluationProofId for HyperKZGCommitmentEvaluationProof {
     const COMMITMENT_SCHEME: CommitmentScheme = CommitmentScheme::HyperKzg;
+    type OwnedVerifierPublicSetup = VerifierKey<HyperKZGEngine>;
+
+    fn deserialize_verifier_setup(bytes: &[u8]) -> Result<VerifierKey<HyperKZGEngine>, String> {
+        bincode::serde::decode_from_slice(
+            bytes,
+            bincode::config::legacy()
+                .with_fixed_int_encoding()
+                .with_big_endian(),
+        )
+        .map(|(setup, _)| setup)
+        .map_err(|e| format!("Failed to deserialize verifier setup: {}", e))
```

NIT: This line is the same in both implementations. Is there ever going to be a case where this isn't the final step? If not, I'd just return the error message in this function. Though maybe you want to say the commitment scheme in your error message.